T_run_mcl.py

```diff
@@ -41,18 +41,24 @@ def __init__(self, ax, global_map, particle_list, target_particles=300,  draw_ma
         self.resample_period = resample_period
 
     def update(self, message):
-        print(self.i, end='\r') 
-        sys.stdout.flush()
         if self.i % self.resample_period == 0:# Resample and plot state
             self.particle_list = mcl.mcl_update(self.particle_list, message, resample=True,
                                                 target_particles=self.target_particles) # Update
-            plt.cla()        
+            plt.cla()
+            sum_x = sum_y = 0
+            for p in self.particle_list:
```

Comment:
I don't think we can avoid the for loop nature of this, but we can at least do it on the numpy side, and get it to vectorize the x and y values, which we can feed to numpy's mean funciton:

see https://stackoverflow.com/questions/54199914/access-attribute-of-elements-within-numpy-array for an example of someone doing it, and here for the vectorize docs: https://docs.scipy.org/doc/numpy-1.9.2/reference/generated/numpy.vectorize.html
e.g.
``` python
a = np.array([dict(pose=[1,2]), dict(pose=[3,4]), dict(pose=[5,6]), dict(pose=[7,8])])
poses = np.vectorize(lambda x: (x['pose'][0], x['pose'][1]))(a)
avg_x = np.mean(poses[0])
avg_y = np.mean(poses[1])
```

obviously for this use case it would be `np.vectorize(lambda x: (x['pose'][0], x['pose'][1]))` because `pose` is an object not a dict, but you get the idea.